File: java/src/main/java/com/penapereira/patterns/interpreter/NumberExpression.java
package com.penapereira.patterns.interpreter;

/** Terminal: a literal number. */
public final class NumberExpression implements AbstractExpression {

    private final int value;

    public NumberExpression(int value) {
        this.value = value;
    }

    @Override
    public int interpret(Context context) {
        return value;
    }

    @Override
    public String describe() {
        return Integer.toString(value);
    }
}

File: typescript/src/interpreter/number-expression.ts
import type { AbstractExpression } from "./abstract-expression.ts";

/** Terminal: a literal number. */
export class NumberExpression implements AbstractExpression {
  constructor(private readonly value: number) {}

  interpret(): number {
    return this.value;
  }

  describe(): string {
    return String(this.value);
  }
}
